File: Optimal/Cpp/q344_reverseString.cpp
class Solution {
    public:
        void reverseString(vector<char>& s) {
            
            int front=0,back=s.size()-1;
    
            while(front<back){
                swap(s[front],s[back]);
                front++;
                back--;
            }
        }
    };

//2 pointer approach, beats 100% timewise

File: MySolutions/Python/q344_reverseString.py
#Ideal approach using 2 pointer technique

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        l=len(s)
        front=0
        back=l-1

        while front<back:
            s[front],s[back]=s[back],s[front]
            front+=1
            back-=1